content/blog/2017/08/2017-08-17-speaker-blog-blazemeter.adoc

```diff
@@ -0,0 +1,177 @@
++
+[source, yaml]
+----
+## execution:
+- concurrency: 100
+  hold-for: 10m
+  ramp-up: 120s
+  scenario: Thread Group
+scenarios:
+  Thread Group:
+    requests:
+    - label: blazedemo
+      method: GET
+      url: http://blazedemo.com/
+----
++
+This script runs 100 concurrent users, holds the load for 10 minutes, the
+ramp-up is 120 seconds and the thread group runs one GET request to
+`blazedemo.com`.
++
+You can specify an executor by adding `executor: <executor_name>` to the
+script. Otherwise, the default executor will be JMeter. In the background,
+Taurus will create an artifact directory with a jmx file (or a Scala file if
+you run Gatling, a Python file if you are running Selenium, etc.).
++
+. Open a terminal and run: `bzt <file_name>.yml`
+. View the test results:
+
+
+image::/images/post-images/jenkinsworld2017/blazemeter/blazemeter-speaker-blog-2017-1.png[Viewing test results from Taurus]
+
+If you want to conduct an in-depth analysis of your test results, run your
+tests on BlazeMeter. You will be able to monitor KPIs through advanced and
+colorful reports, evaluate system health over time, and run your tests from
+multiple geo-locations.
+
+Run the following command from the terminal:
+
+[source,bash]
+----
+bzt <file_name>.yml -report`
+----
+
+image::/images/post-images/jenkinsworld2017/blazemeter/blazemeter-speaker-blog-2017-2.png[Viewing test results in Blazemeter]
+
+### Integrate Taurus With Pipeline
+
+To run Taurus through link:/doc/book/pipeline[Pipeline], you can also go
+straight to Jenkins after creating your Taurus script.
+
+. Open Jenkins -> New Item -> Fill in an item name -> Click on ‘Pipeline’
++
+image::/images/post-images/jenkinsworld2017/blazemeter/blazemeter-speaker-blog-2017-3.png[]
++
+image::/images/post-images/jenkinsworld2017/blazemeter/blazemeter-speaker-blog-2017-4.png[]
++
+. Now create a Pipeline script. You can include all parts of
+your CI/CD process in this script: Commit, Build, Unit Test, Performance Test,
+etc., by creating different stages.
++
+This Pipeline has three stages: The first is called “build”. In this example it
+is empty, but you can add commands that will build your code. The second,
+called “Performance Tests”, creates a folder called “Taurus-Repo” and runs the
+Taurus script that we created. At the same time (note the “parallel” command),
+there is a “sleep” command for 60 seconds. Obviously it makes no sense to put
+those two commands together, this is just to show you the option of running 2
+commands in parallel. The third stage called “Deploy” is also empty in this
+example. This is where you could deploy your new version.
++
+[source, groovy]
+----
+node {
+   stage('Build') {
+      // Run the Taurus build
+   }
+   stage('Performance Tests') {
+    parallel(
+        BlazeMeterTest: {
+            dir ('Taurus-Repo') {
+                sh 'bzt <file_name>.yml -report'
+            }
+        }
+        , Analysis: {sleep 60})
+
+   }
+   stage(‘Deploy’) {
```

Comment:
Smart quotes are annoying, but asciidoctor converts them automatically so I didn't bother updating the source material here